nav2_collision_monitor/src/polygon.cpp

```diff
@@ -422,22 +446,48 @@ bool Polygon::getParameters(
         polygon_name_.c_str());
     }
 
-    if (action_type_ == STOP || action_type_ == SLOWDOWN || action_type_ == LIMIT ||
-      action_type_ == DO_NOTHING)
-    {
-      // Dynamic polygon will be used
-      nav2_util::declare_parameter_if_not_declared(
-        node, polygon_name_ + ".polygon_sub_topic", rclcpp::PARAMETER_STRING);
-      polygon_sub_topic =
-        node->get_parameter(polygon_name_ + ".polygon_sub_topic").as_string();
-    } else if (action_type_ == APPROACH) {
-      // Obtain the footprint topic to make a footprint subscription for approach polygon
-      nav2_util::declare_parameter_if_not_declared(
-        node, polygon_name_ + ".footprint_topic",
-        rclcpp::ParameterValue("local_costmap/published_footprint"));
-      footprint_topic =
-        node->get_parameter(polygon_name_ + ".footprint_topic").as_string();
+    try {
```

Comment:
I don't think this handling of try/catch is correct. Why would we assume because there's no subscription topic that it means we're using the velocity polygons?

Reply:
If I understand correctly, in the current design, it is using the try-catch method -- if static points are defined, it will [return early](https://github.com/ros-planning/navigation2/blob/be370db6ea74280eff7b2075b9f31cc6eca56d31/nav2_collision_monitor/src/polygon.cpp#L429C7-L431).  if static points are not defined, it will throw a `ParameterUninitializedException` and the [catch block](https://github.com/ros-planning/navigation2/blob/be370db6ea74280eff7b2075b9f31cc6eca56d31/nav2_collision_monitor/src/polygon.cpp#L432-L437) will log a message and skip to the next part to initialise the parameters for dynamic subscriptions.

In the changes here, I follow the same pattern -- if `polygon_sub_topic` or `footprint_topic`  is defined, it will return early. if `polygon_sub_topic` or `footprint_topic` is not defined, the catch block will skip to the next part to initialise the parameters for velocity-based polygon.

Reply:
This logic needs to be adjusted to check the nature of the static footprint information -- whether we have a static single footprint or we have the parameters associated with the velocity polygons to use. If neither of those are true, then we fail. We can check this explicitly. 

I think its worth from this pause in the development of this feature to take a pause and potentially rethink areas like this that with the new feature might just need to be refactored how they work now that we're adding in a 3rd modality. Think more openly about the best way to do this explicitly, not relying on cascading failures. We know what params might be filled in for each mode -- we should be able to check which mode is set by them (and if multiple params set making ambiguous, fail the configuration)